### PR TITLE
Add $uploadResult variable to JFile class

### PR DESCRIPTION
**Pull request by olleharstedt**
Pull Request for Issue #20246.

### Summary of Changes

Add static variable `$uploadResult` to JFile class.

### Testing Instructions

Run JFile::upload(). Run JFile::getUploadResult().

### Expected result

$uploadResult is set to result of upload() method.

### Documentation Changes Required

Don't know.

### CODE REVIEW

**Comment by PhilETaylor** [comment]
Why are you declaring new constants - can we not keep compatibility/same as http://php.net/manual/en/features.file-upload.errors.php ??

**Comment by olleharstedt** [comment]
Hm, those constants does not really reflect the error messages logged in upload().

**Comment by PhilETaylor** [comment]
Ah ok  - cool then

**Comment by olleharstedt** [comment]
Something that could be discussed is if we should use the convention "0 for success" instead of "truthy for success", as I've used now (the constants start at 1 and then counts down).

File: libraries/joomla/filesystem/file.php [hunk 1]
```diff
@@ -657,4 +700,14 @@ public static function getName($file)
 			return $file;
 		}
 	}
+
+	/**
+	 * Return result of last upload() method call.
+	 *
+	 * @return int
+	 */
```
Comment (Quy) [inline libraries/joomla/filesystem/file.php]: Spell out `int`. Add `* @since   __DEPLOY_VERSION__`

File: libraries/joomla/filesystem/file.php [hunk 2]
```diff
@@ -16,6 +16,42 @@
  */
 class JFile
 {
+	/**
+	 * File uploaded successfully.
+	 * @var int
```
Comment (Quy) [inline libraries/joomla/filesystem/file.php]: Spell out `int`. Add `* @since __DEPLOY_VERSION__`. Repeat below.

**Comment by laoneo** [comment]
Should we not start throwing exceptions? I don't like to have state full helper classes. It is not wrong how you do it, but we can do better.

**Comment by laoneo** [comment]
Hit the "Comment" button too early. Yes then we should do it against J4.

**Comment by olleharstedt** [comment]
I agree that throwing exception is the correct way, but yes, it's about backwards compatibility. Alternatively, we can have a new method `uploadWithException` or something, that does throw exception (code duplication, though...).

**Comment by mbabker** [comment]
Instead of trying to cater for B/C, do it once and do it right.  Do a PR against 4.0 (and the 2.0 branch of the Framework's Filesystem package) breaking B/C to do it right.  Yeah it means you have to live with the "busted" code for the rest of the 3.x lifetime, but at this point it's better than adding something that creates new stateful statics or APIs which are nothing more than complete hacks that are going to have to stay in the API for who knows how long.

**Comment by olleharstedt** [comment]
How about changing the return value from true/false to 1 and 0/-1/-2 etc as return codes? It would break compatibility, but not "so much" (only if client code uses `===`).

**Comment by ghost** [comment]
@olleharstedt whats the state of this Pull Request?

**Comment by olleharstedt** [comment]
@franz-wohlkoenig It should be closed, and a discussion should be had about my other suggestion: to return integer instead of boolean as a result code OR throw exception in version 4 instead.

**Comment by ghost** [comment]
thanks @olleharstedt - can you open an issue about "return integer instead of boolean"?
closed as stated above.

**Comment by olleharstedt** [comment]
Sure, I'll have a look tomorrow. :+1:

**Pull request closed without merging**